backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/po/CustomColumnDAO.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * #%L
+ * de.metas.adempiere.adempiere.base
+ * %%
+ * Copyright (C) 2022 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.po;
+
+import com.google.common.collect.ImmutableMap;
+import de.metas.i18n.AdMessageKey;
+import de.metas.util.converter.POValueConverters;
+import lombok.NonNull;
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.compiere.model.PO;
+import org.compiere.model.POInfo;
+import org.compiere.model.POInfoColumn;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nullable;
+import java.time.ZoneId;
+import java.util.Map;
+import java.util.Optional;
+
+@Component
+public class CustomColumnDAO
```

Comment:
What's the reason of splitting out CustomColumnDAO from CustomColumnService?

I observe that in DAO you are updating the PO and saving it,
but in Service you are loading it from TableRecordReference.

So i am wonder what's the responsibility of DAO and what's the responsibility of the Service?

Reply:
The idea behind was to have the Service put all the pieces together, i.e. grabbing the zoneId, and the DAO to only be concerned about setting some custom values on a given PO taking into consideration a given Org :-?? but indeed, we could also drop the service layer and stick only with the DAO